lightning/src/ln/interactivetxs.rs

```diff
@@ -0,0 +1,667 @@
+		let get_output = |outpoint: &OutPoint| {
+			if outpoint.txid == tx_to_validate.txid() {
+				return tx_to_validate.output.get(outpoint.vout as usize).cloned()
+			} else {
+				None
+			}
+		};
+		if let Err(_) = tx_to_validate.verify(get_output) {
+			return Err(AbortReason::InvalidTransactionState)
+		};
+
+		// - there are more than 252 inputs
+		// - there are more than 252 outputs
+		if self.context.inputs.len() > MAX_INPUTS_OUTPUTS_COUNT ||
+			self.context.outputs.len() > MAX_INPUTS_OUTPUTS_COUNT {
+			return Err(AbortReason::ExceededNumberOfInputsOrOutputs)
+		}
+
+		if tx_to_validate.weight() > MAX_STANDARD_TX_WEIGHT {
+			return Err(AbortReason::TransactionTooLarge)
+		}
+
+		// TODO: Need to figure out how to do this
+		// if is the non-initiator:
+		// 	- the initiator's fees do not cover the common fields (version, segwit marker + flag,
+		// 		input count, output count, locktime)
+
+		return Ok(())
+	}
+
+	fn is_valid_counterparty_serial_id(&self, serial_id: SerialId) -> bool {
+		// A received `SerialId`'s parity must match the role of the counterparty.
+		self.context.holder_is_initiator == !serial_id.is_valid_for_initiator()
+	}
+}
+
+impl InteractiveTxStateMachine<TheirTxComplete> {
+	fn send_tx_complete(self) -> InteractiveTxStateMachine<NegotiationComplete> {
+		// TODO: Should we validate before transitioning states? If so, do we want to abort negotiation
+		// if our current transaction state is invalid?
+		InteractiveTxStateMachine {
+			context: self.context,
+			state: NegotiationComplete {}
+		}
+	}
+}
+
+impl InteractiveTxStateMachine<Negotiating> {
+	fn receive_tx_complete(self) -> Result<InteractiveTxStateMachine<TheirTxComplete>, InteractiveTxStateMachine<NegotiationAborted>> {
+		match self.is_current_transaction_state_able_to_complete() {
+			Err(e) => Err(InteractiveTxStateMachine { context: self.context, state: NegotiationAborted(e) }),
+			_ => Ok(InteractiveTxStateMachine {
+				context: self.context,
+				state: TheirTxComplete {}
+			})
+		}
+	}
+
+	fn send_tx_complete(self) -> InteractiveTxStateMachine<OurTxComplete> {
+		// TODO: Should we validate before transitioning states? If so, do we want to abort negotiation
+		// if our current transaction state is invalid?
+		InteractiveTxStateMachine {
+			context: self.context,
+			state: OurTxComplete {}
+		}
+	}
+}
+
+impl InteractiveTxStateMachine<OurTxComplete> {
+	fn receive_tx_complete(self) -> Result<InteractiveTxStateMachine<NegotiationComplete>, InteractiveTxStateMachine<NegotiationAborted>> {
+		match self.is_current_transaction_state_able_to_complete() {
+			Err(e) => Err(InteractiveTxStateMachine { context: self.context, state: NegotiationAborted(e) }),
+			_ => Ok(InteractiveTxStateMachine {
+				context: self.context,
+				state: NegotiationComplete {}
+			})
+		}
+	}
+}
+
+impl InteractiveTxStateMachine<NegotiationComplete> {
+	fn get_psbt(&self) -> Result<Transaction, AbortReason> {
```

Comment:
you have a PSBT support in rust-bitcoin, see `bitcoin/src/psbt/mod.rs` if needed.